docs/api/ipc-renderer.md

```diff
@@ -57,11 +57,41 @@ Removes all listeners, or those of the specified `channel`.
 * `...args` any[]
 
 Send a message to the main process asynchronously via `channel`, you can also
-send arbitrary arguments. Arguments will be serialized in JSON internally and
+send arbitrary arguments. Arguments will be serialized as JSON internally and
 hence no functions or prototype chain will be included.
 
 The main process handles it by listening for `channel` with [`ipcMain`](ipc-main.md) module.
 
+### `ipcRenderer.invoke(channel[, arg1][, arg2][, ...])`
+
+* `channel` String
+* `...args` any[]
+
+Returns `Promise<any>` - Resolves with the response from the main process.
+
+Send a message to the main process asynchronously via `channel` and expect an
+asynchronous result. Arguments will be serialized as JSON internally and
+hence no functions or prototype chain will be included.
+
+The main process should listen for `channel` with the [`ipcMain`](ipc-main.md)
+module, and send a result by calling `event.reply()`.
```

Comment:
This leaves a few unanswered questions:
* What happens when `event.reply()` is called multiple times?
* What happens if multiple listeners for the `channel` are added in the main process

It isn't quite as nice an API as `ipcMainInternal.handle(channel)` which automatically proxies errors thrown in the handler.  I'm imaginging:

```js
// Will throw / warn if a handler for "channel" is already set
ipcMain.setHandler('channel', async () => {
  return 'myvalue';
});
```

Reply:
the 2nd and subsequent `event.reply()` calls will be no-ops (the callback is stored as an `Optional<OnceCallback>` and it gets `std::move`d to run, with a check to make sure it's present).

Maybe they should error instead?

Multiple listeners = all of them get called, whichever one calls `event.reply` first wins. I don't think it's especially useful to have multiple listeners, but I also don't see a particular reason to go through any special pains to disallow it.

Reply:
> Multiple listeners = all of them get called, whichever one calls event.reply first wins

This sounds like a foot-gun that we should prevent users experiencing